frontend/src/components/BCDataGrid/BCDataGridEditor.jsx

```diff
@@ -0,0 +1,205 @@
+      rowHeight: 45,
+      headerHeight: 40,
+      animateRows: true,
+      suppressPaginationPanel: true,
+      suppressScrollOnNewData: true,
+      suppressCsvExport: false,
+      components: frameworkComponents,
+      onFirstDataRendered: (params) => {
+        params.api?.setFocusedCell(0, params.api?.getColumns()[2].getColId())
+
+        params.api?.startEditingCell({
+          rowIndex: 0,
+          colKey: params.api?.getColumns()[2].getColId()
+        })
+      },
+      // stopEditingWhenCellsLoseFocus: true,
+      getRowStyle: highlightedRowId
+        ? (params) => {
+            if (params.node.id === highlightedRowId) {
+              return { backgroundColor: '#fade81' }
+            }
+          }
+        : undefined
+    }),
+    [highlightedRowId]
+  )
+
+  // Function to add a new row
+  const addRow = useCallback(() => {
+    const id = uuid()
+    const emptyRow = { id, modified: true }
+    gridApi.applyTransaction({ add: [emptyRow] })
+  })
+
+  // Function to delete selected row
+  const deleteRow = useCallback((props) => {
+    const selectedData = columnApi?.api?.getSelectedRows()
+    gridApi?.applyTransaction({ remove: selectedData })
+  })
+
+  // Function called when cell value changes
+  function onCellValueChanged(event) {
+    console.log('onCellValueChanged', event)
+    event.data.modified = true // Mark the entire row as modified
+  }
+
+  return (
+    <BCBox
+      sx={{
+        width: '100%',
+        height: '100%',
+        display: 'flex',
+        flexDirection: 'column'
+      }}
+      className="bc-grid-container"
+    >
+      <AgGridReact
+        gridRef={gridRef}
+        gridApi={gridApi}
+        className={className}
+        columnDefs={columnDefs}
+        defaultColDef={defaultColDef}
+        rowData={rowData}
+        getRowNodeId={getRowNodeId}
+        gridOptions={{ ...defaultGridOptions }}
+        onGridReady={onGridReady}
+        frameworkComponents={frameworkComponents}
+        domLayout="autoHeight"
+        onCellValueChanged={onCellValueChanged}
+        editType={'fullRow'}
+        {...others}
+      />
+      <BCBox
+        display="flex"
+        justifyContent="flex-start"
+        variant="outlined"
+        sx={{ maxHeight: '4.5rem', position: 'relative' }}
+      >
+        <Stack spacing={2} direction="row" m={2}>
+          <BCButton variant="contained" color="primary" onClick={saveData}>
+            Save changes
+          </BCButton>
+          <BCButton variant="outlined" color="primary" onClick={addRow}>
+            Add new row
+          </BCButton>
+          <BCButton variant="outlined" color="error" onClick={deleteRow}>
+            Delete selected rows
+          </BCButton>
+        </Stack>
+      </BCBox>
+    </BCBox>
+  )
+}
+
+BCDataGridEditor.defaultProps = {
+  gridRef: null,
+  gridKey: `bcgrid-key-<unique-id>`,
+  gridOptions: {},
+  loadingOverlayComponentParams: { loadingMessage: 'One moment please...' },
```

Comment:
is this used?

Reply:
yes, I've enabled it in ag-grid now. It would useful in loading inital rows or during saving changes.